src/internal/m365/collection/site/collection.go

```diff
@@ -358,7 +358,7 @@ func (sc *Collection) handleListItems(
 	rc := io.NopCloser(bytes.NewReader(entryBytes))
 	itemInfo := details.ItemInfo{
 		SharePoint:     info,
-		NotRecoverable: template == api.WebTemplateExtensionsListTemplateName,
+		NotRecoverable: api.NonRestorableListTemplates.HasKey(template),
```

Comment:
I guess this could have been two ifs.

Reply:
a list entry needs to be NonRecoverable if the template is either 'webTemplateExtensionsList' or 'documentLibrary'(temporarily). And NonRestorableListTemplates map contains both of them, so if the current listTemplate is either of the two types, NotRecoverable becomes true for that entry.

Reply:
Make sense, I was mentioning that this could be `template == api.WebTemplateExtensionsListTemplateName || template == api.documentLibrary`. I personally feel this to be more clear.